include/zephyr/bluetooth/iso.h

```diff
@@ -621,10 +621,121 @@ int bt_iso_chan_disconnect(struct bt_iso_chan *chan);
  */
 int bt_iso_chan_send(struct bt_iso_chan *chan, struct net_buf *buf);
 
+struct bt_iso_unicast_tx_info {
+	/** The transport latency in us */
+	uint32_t latency;
+
+	/** The flush timeout in ms */
+	uint32_t flush_timeout;
+
+	/** The transport PHY  */
+	uint8_t  phy;
+
+	/** The burst number */
+	uint8_t  bn;
+
+	/** The maximum PDU size in octets */
+	uint16_t max_pdu;
+};
+
+struct bt_iso_unicast_info {
+	/** The maximum time in us for all PDUs of all CIS in a CIG event */
+	uint32_t cig_sync_delay;
+
+	/** The maximum time in us for all PDUs of this CIS in a CIG event */
+	uint32_t cis_sync_delay;
+
+	/** The maximum number of subevents in each ISO event */
+	uint8_t  max_subevent;
+
+	/** The ISO interval in ms */
+	uint16_t interval;
+
+	/** @brief TX information when sending from central to peripheral
+	 *
+	 *  All values will be 0 if the central cannot send to the peripheral.
+	 */
+	struct bt_iso_unicast_tx_info central;
+
+	/** TX information when sending from peripheral to central
+	 *
+	 *  All values will be 0 if the peripheral cannot send to the central.
+	 */
+	struct bt_iso_unicast_tx_info peripheral;
+};
+
+struct bt_iso_broadcaster_info {
+	/** The maximum time in us for all PDUs of all BIS in a BIG event */
+	uint32_t sync_delay;
+
+	/** The transport latency in us */
+	uint32_t latency;
+
+	/** The transport PHY  */
+	uint8_t  phy;
+
+	/** The maximum number of subevents in each ISO event */
+	uint8_t  max_subevent;
+
+	/** The burst number */
+	uint8_t  bn;
+
+	/** Number of times a payload is transmitted in a BIS event */
+	uint8_t  irc;
+
+	/** Pre-transmission offset in ms */
+	uint32_t  pto;
+
+	/** The maximum PDU size in octets */
+	uint16_t max_pdu;
+
+	/** The ISO interval in ms */
+	uint16_t interval;
+};
+
+struct bt_iso_sync_receiver_info {
+	/** The transport latency in us */
+	uint32_t latency;
+
+	/** The maximum number of subevents in each ISO event */
+	uint8_t  max_subevent;
+
+	/** The burst number */
+	uint8_t  bn;
+
+	/** Number of times a payload is transmitted in a BIS event */
+	uint8_t  irc;
+
+	/** Pre-transmission offset in ms */
+	uint32_t  pto;
+
+	/** The maximum PDU size in octets */
+	uint16_t max_pdu;
+
+	/** The ISO interval in ms */
+	uint16_t interval;
```

Comment:
Values such as PHY, BN and interval are common between the 3 structs. Should we move them out of these and store them directly in `struct bt_iso_info`?

Reply:
I like that - then you always have the values available where you need them (in the data path) and don't need for each application to implement it's own mapping from connection instance to connection parameters.
But again the unit of interval shall support x.25ms and x.5ms hence perhaps use micro seconds?

Reply:
See https://github.com/zephyrproject-rtos/zephyr/pull/45077#discussion_r857551073

Reply:
I move the NSE and interval to the outer struct, but the remaining fields are not easily moved. While all have e.g. max_pdu, the unicast struct have 2 values for that (given that it can be bidirectional). 

The unicast and broadcast data path values (PDU, bn, latency, etc.) are also not the same, as e.g. broadcast has PTO and unicast has flush timeout